### PR TITLE
chore: Exclude native vscode.github-authentication from extension dependencies

### LINKED ISSUE #21492: Not possible to sign in with GitHub from GitHub Pull Requests and Issues extension in Che-Code
### Describe the bug

After starting a workspace with Che-Code and GitHub Pull Requests and Issues extension, we have to sign in into GitHub to continue work with GitHub Pull Requests extension but sign in process is failed.

### Che version

next (development version)

### Steps to reproduce

1. Start a workspace via factory with project https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#/https://github.com/svor/quarkus-api-example/tree/sv-che-code
2. Install recommended GitHub Pull Requests and Issues extension
3. Go to GitHub view and click Sign in button 

### Expected behavior

Che-Code should use our OAuth provider to authenticate in GitHub

### Runtime

OpenShift

### Screenshots

screenshot-che-dogfooding apps che-dev x6e0 p1 openshiftapps com-2022 06 24-13_04_15

screenshot-vscode-auth github com-2022 06 24-13_09_59

### Installation method

OperatorHub

### Environment

Linux

### PR DESCRIPTION
**Pull request by vinokurig**
* Ignore the native `github-authentication` extension from extension dependencies to not coliede with `che-github-authentication` extension which overrides the native one.
* Fix the credentials file location.
fixes https://github.com/eclipse/che/issues/21492, https://github.com/eclipse/che/issues/21443

### CODE REVIEW

File: code/extensions/che-api/src/impl/github-service-impl.ts [hunk 1]
```diff
@@ -14,15 +14,13 @@ import { GithubService, GithubUser } from '../api/github-service';
...
-import * as path from 'path';
-import * as os from 'os';
 
 @injectable()
 export class GithubServiceImpl implements GithubService {
     private readonly token: string | undefined;
 
     constructor(@inject(Symbol.for('AxiosInstance')) private readonly axiosInstance: AxiosInstance) {
-        const credentialsPath = path.resolve(os.homedir(), '.git-credentials', 'credentials');
+        const credentialsPath = '/.git-credentials/credentials';
```
Comment (azatsarynnyy) [inline code/extensions/che-api/src/impl/github-service-impl.ts]: Why not `path.resolve('/.git-credentials', 'credentials')`?
Yes, we're always in the Linux environment, but it's a more true way )

File: code/src/vs/workbench/api/common/extHostExtensionActivator.ts [hunk 2]
```diff
@@ -170,6 +170,9 @@ export class ExtensionsActivator implements IDisposable {
 	private readonly _externalExtensionsMap: Map<string, ExtensionIdentifier>;
 	private readonly _host: IExtensionsActivatorHost;
 	private readonly _operations: Map<string, ActivationOperation>;
+	private readonly _excludedDependencies = [
+		'vscode.github-authentication',
```
Comment (azatsarynnyy) [inline code/src/vs/workbench/api/common/extHostExtensionActivator.ts]: Does it make the previously added exclusion mechanism redundant?
https://github.com/che-incubator/che-code/commit/7aa9f31489c0dcbe38841e4204770aa6b6128bf2#diff-36fae229a2323a58cfac9a248b606846fd67fbf795dd4f8aecacfd6007faf134R416

**Comment by azatsarynnyy** [comment]
For testing, I've used [the fork](https://github.com/azatsarynnyy/quarkus-api-example/tree/sv-che-code) of the [test repo]([https://github.com/svor/quarkus-api-example/tree/sv-che-code](https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#/https://github.com/svor/quarkus-api-example/tree/sv-che-code)) given in the issue https://github.com/eclipse/che/issues/21492

I've configured it to use the Che-Code image built from this PR
https://github.com/azatsarynnyy/quarkus-api-example/blob/52999f1fc90a719a231bec7137528d311ddf8da7/.che/che-editor.yaml#L65

It looks like the built-in `github-authentication` extension is not excluded properly:
image

so I've got the expected error:
image

@vinokurig could you look at it? Maybe something is wrong with the steps I've used for testing...

File: code/src/vs/workbench/api/common/extHostExtensionActivator.ts (hunk 2 above)
Comment (vinokurig) [reply code/src/vs/workbench/api/common/extHostExtensionActivator.ts]: No, connected it with the new check.

**Comment by vinokurig** [comment]
@azatsarynnyy Fixed and reworked the PR, please remember to uninstall the `github-pull-request` and `github-authentication` extensions before testing. Thank you.

**Comment by azatsarynnyy** [comment]
Thanks, I'll check it today once the dogfooding instance is fixed. Or on Monday on my local cluster.

File: code/extensions/che-api/src/impl/github-service-impl.ts (hunk 1 above)
Comment (vinokurig) [reply code/extensions/che-api/src/impl/github-service-impl.ts]: done

**Review by azatsarynnyy — APPROVED** [review]
I'm able to sign in to GitHub with the proposed changes.

**Pull request merged**